### PR TITLE
Refactor OmniSci to Heavy

### LINKED ISSUE #451: Rename omnisci to heavydb
With https://github.com/xnd-project/rbc/pull/447, rbc supports two brands of heavydb:
- heavyai: version 6.0 and up
- omnisci: versions 5.x and older

In long term (1-2 years), omnisci brand is going to be deprecated and unsupported.

In short term, we'll need to switch to using heavyai brand name instead of omnisci brand name while still supporting the omniscidb 5.7 and up.

The switch means renaming heavydb related directories and file names as follows [this is a suggestion to be discussed]:
```
rbc/omniscidb.py -> rbc/heavydb.py
rbc/omnisci_backend/ - rbc/heavyai/
rbc/omnisci_backend/omnisci_xyz.py -> rbc/heavyai/xyz.py
rbc/tests/test_omnisci_xyz.py -> rbc/tests/heavyai/test_xyz.py
rbc/externals/omniscidb.py -> rbc/externals/heavydb.py
```

This issue will be actionable after heavydb 6.0 is released and heavydb conda packages will be available.

### PR DESCRIPTION
**Pull request by tupui**
Closes #451

### CODE REVIEW

File: rbc/heavyai/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+from .bytes import *  # noqa: F401, F403
+from .metatype import *  # noqa: F401, F403
+from .text_encoding import *  # noqa: F401, F403
+from .pipeline import *  # noqa: F401, F403
+from .column_list import *  # noqa: F401, F403
+from .table_function_manager import *  # noqa: F401, F403
+
+import rbc.heavyai.mathimpl as math  # noqa: F401
+import rbc.heavyai.npyimpl as np  # noqa: F401
+import rbc.heavyai.python_operators as operators  # noqa: F401
```
Comment (pearu) [inline rbc/heavyai/__init__.py:12]:
```suggestion
import .mathimpl as math  # noqa: F401
import .npyimpl as np  # noqa: F401
import .python_operators as operators  # noqa: F401
```

File: rbc/omniscidb.py [hunk 2]
```diff
@@ -1,1479 +1,6 @@
...
-        if is_udtf_call:
-            return numpy.array(list(result), dtype).view(numpy.recarray)
-        else:
-            return dtype[0][1](list(result)[0][0])
-
-
-class RemoteOmnisci(RemoteHeavyDB):
-    """Omnisci - the previous brand of HeavyAI
-    """
+msg = "`RemoteOmnisci` is deprecated, use `RemoteHeavyDB` instead."
```
Comment (pearu) [inline rbc/omniscidb.py:5]:
```suggestion
msg = "`import rbc.omniscidb` is deprecated, use `import rbc.heavyai` instead."
```

File: rbc/remotejit.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 # XXX WIP: the OmnisciCompilerPipeline is no longer omnisci-specific because
 # we support Arrays even without omnisci, so it must be renamed and moved
 # somewhere elsef
-from .omnisci_backend import OmnisciCompilerPipeline
+from .heavyai import OmnisciCompilerPipeline
```
Comment (pearu) [inline rbc/remotejit.py:22]: At some point (not necessarily here to keep this PR simple), we'll need to rename `OmnisciCompilerPipeline` as well.

**Review by pearu — CHANGES_REQUESTED** [review]
Thanks, @tupui for this work!

I have a couple of nits.

In addition, at some point, we should filter out heavydb dependent code from test_externals_cmath.py and test_externals_libdevice.py as well and move these under tests/heavyai/

**Comment by pearu** [comment]
https://github.com/xnd-project/rbc/blob/master/.github/workflows/rbc_test.yml requires an update as well. For instance, currently, CI does not run any of the heavyai tests..

**Comment by tupui** [comment]
Most tests seem to be passing except: `rbc/tests/heavyai/test_math.py::test_numpy_function[np.positive-cpu]`
There is a core dump when evaluating: `nb.njit(fn)`

**Comment by tupui** [comment]
ok we debugged with @guilhermeleobas and #445 fixes the segfault.

@pearu Once this is green, I propose to merge maybe with the test skip, and then @guilhermeleobas would remove it in his PR.

**Comment by tupui** [comment]
> Considering that this PR is mostly about renaming omnisci to heavydb, @tupui , @guilhermeleobas could you explain how #445 that deals with numba internals is related to the changes in this PR?

AFAIU, @guilhermeleobas noticed it elsewhere.

**Comment by tupui** [comment]
Ok seems like all CI is passing. There is one workflow failing one test but seems unrelated `test_add_ints_rjit`. I tried to relaunch the failed workflow but GitHub is having issues currently.

**Comment by tupui** [comment]
See here for to view the previous run: https://github.com/xnd-project/rbc/actions/runs/2023309018/attempts/1

**Comment by tupui** [comment]
@pearu @guilhermeleobas the CI is green. It was just a hiccup (maybe even related to GH's current outage).

**Review by guilhermeleobas — APPROVED** [review]

**Review by pearu — APPROVED** [review]
LGTM! There are a couple of discussion items that should be addressed in a follow-up but let's land this now.

Thanks, @tupui!

File: .github/workflows/rbc_test.yml [hunk 4]
```diff
@@ -225,7 +225,7 @@ jobs:
           EXPECTED_NUMBA_VERSION: ${{ matrix.numba-version }}
           RBC_TESTS_FULL: TRUE
         run: |
-          mamba run -n rbc pytest -sv -r A rbc/tests/ -x -k omnisci
+          mamba run -n rbc pytest -sv -r A rbc/tests/ -x -k heavyai
```
[the remark this answers is not in the document]
Comment (tupui) [reply .github/workflows/rbc_test.yml:228]: Sure 👍

**Pull request merged**